tests/scripts/analyze_outcomes.py

```diff
@@ -85,7 +85,12 @@ def execute_reference_driver_tests(results: Results, ref_component: str, driver_
 def analyze_coverage(results: Results, outcomes: Outcomes,
                      allow_list: typing.List[str], full_coverage: bool) -> None:
     """Check that all available test cases are executed at least once."""
-    available = check_test_cases.collect_available_test_cases()
+    try:
+        available = check_test_cases.collect_available_test_cases()
+    except check_test_cases.ScriptOutputError:
+        results.error("fail to collect available test cases")
```

Comment:
Catching the exception here completely hides the reason why running the script failed. That will make debugging hard if something goes wrong. Especially if something goes wrong on the CI and it's not trivially reproducible locally (e.g. python version dependency, disk full...).

I think that collecting test cases is a required phase, and we can treat errors during that phase as fatal. If we don't even know what test cases are available, we can't do any meaningful analysis. So just let this exception abort the program. This is different from scenarios that can happen if something went wrong while running the tests, then we want to report as much as possible about the tests that did run and so we use the `results` class to record observations.

Ad hoc tests:

* If you remove `tests/compat.sh` and run `analyze_outcomes.py`, there should be an error message conveying that `tests/compat.sh` is missing.
* If you replace the contents of `tests/compat.sh` by `echo something is wrong`, there should be an error message conveying that `tests/compat.sh` produced output with an invalid format.